### DISCUSSION TITLE
Next.JS Twemoji image duplication bug

### DISCUSSION THREAD

**Discussion by cliid** [vercel/next.js#29737] [Help]
My application works (what I mean by 'work' is that it doesn't have the bug) locally but not when deployed to Vercel. My website is [https://cliid.dev](https://cliid.dev) and I'm using [twemoji](https://github.com/twitter/twemoji.git) client side, due to the fact that twemoji uses DOM walking. I tried to find other Next.JS compatible twemoji solutions, but I couldn't find any. By the way, I'm using `next/link` for client-side routing. 

Anyway, I imported twemoji like this: 
```ts
const { default: twemoji } = await import('twemoji');
twemoji.parse(document.body);
```

and also tried this way:

```ts
import twemoji from 'twemoji'

twemoji.parse(document.body);
```

I also thought if this bug was happening because of the client-server html mismatch when the routing occurs, but I couldn't go further.

Oh, I'll also add the full `useEffect` hook that I used.

```ts
useEffect(() => {
    const parseTwemoji = async () => {
      const { default: twemoji } = await import('twemoji');
      twemoji.parse(document.body);
    };
    parseTwemoji();
});
```

**Comment by cliid** [comment]
Oh, the Next.js docs says that `Any <Link /> in the viewport (initially or through scroll) will be prefetched by default (including the corresponding data) for pages using Static Generation.` So I think this is the issue, maybe, because the non-twemojified version gets prefetched and some mismatch happens...?

**Reply by TheCukitoDev** [reply]
Hello, i'm trying to implement twemoji in my web because i don't like unicode. Did you solved it? And how?
Thanks
